lib/browser/api/browser-window.js

```diff
@@ -74,109 +72,85 @@ BrowserWindow.prototype._init = function () {
 
   // Subscribe to visibilityState changes and pass to renderer process.
   let isVisible = this.isVisible() && !this.isMinimized()
-  let visibilityChanged = () => {
-    let newState = this.isVisible() && !this.isMinimized()
+  const visibilityChanged = () => {
+    const newState = this.isVisible() && !this.isMinimized()
     if (isVisible !== newState) {
       isVisible = newState
       this.webContents.send('ELECTRON_RENDERER_WINDOW_VISIBILITY_CHANGE', isVisible ? 'visible' : 'hidden')
     }
   }
-  this.on('show', visibilityChanged)
-  this.on('hide', visibilityChanged)
-  this.on('minimize', visibilityChanged)
-  this.on('restore', visibilityChanged)
-  this.on('maximize', visibilityChanged)
+
+  const visibilityEvents = ['show', 'hide', 'minimize', 'maximize', 'restore']
+  for (let event of visibilityEvents) {
+    this.on(event, visibilityChanged)
+  }
 
   // Notify the creation of the window.
   app.emit('browser-window-created', {}, this)
 
   Object.defineProperty(this, 'devToolsWebContents', {
     enumerable: true,
     configurable: false,
-    get: function () {
+    get () {
       return this.webContents.devToolsWebContents
     }
   })
 }
 
-BrowserWindow.getFocusedWindow = function () {
-  var i, len, window, windows
-  windows = BrowserWindow.getAllWindows()
-  for (i = 0, len = windows.length; i < len; i++) {
-    window = windows[i]
-    if (window.isFocused()) {
-      return window
-    }
+BrowserWindow.getFocusedWindow = () => {
+  for (let window of BrowserWindow.getAllWindows()) {
+    if (window.isFocused()) return window
   }
-  return null
```

Comment:
This is actually needed since this function needs to return null when there is no focused window.